src/components/Calculator.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import './style.css';
+import { ResultIcone } from './resultIcone';
+import { LigneOfButton } from './ligneOfButton';
+import calculate from '../logic/calculate';
+class CalculatorDesign extends React.Component {
+  constructor(props){
+    super(props)
+    this.state = {};
+    this.handleClick = this.handleClick.bind(this);
+    this.prev = null;
+  }
+  handleClick (e){
+    console.log('clicked...')
+    const target = e.target;
+    const child = target.children;
+    let targetElem;
+    if (child.length > 0){
+      targetElem = child[0];
+    }else {
+      targetElem = target;
+    }
+    const buttonName = targetElem.textContent;
+    console.log(buttonName);
+    this.setState(dataObject =>{
+      const newState = calculate(dataObject,buttonName);
+      return newState;
+    });
+  }
+  
+  render(){
+    let res;
+    if(Object.keys(this.state).length ===0){
+      res = 0;
+    }else{
+      if (this.state.total){
+        if(!this.state.operation){
+          res = this.state.total;
+        }else{
+          if(!this.state.next){
+            if(!this.prev){
+              res = this.state.operation;
+            }else{
+              res = this.state.total;
+            }
+          }else{
+            res = this.state.next;
+          }
+
+        }
+      }else {
+        if(this.state.operation){
+          res = this.state.operation;
+        }else {
+          if(this.state.next){
+            res = this.state.next;
+          }else{
+            res = 0;
+          }
+        }
+      }
+    }
+    this.prev = this.state.next;
+    /*let result; 
+    if(Object.keys(this.state).length === 0){
+      result = 0;
+    }else {
+      if (this.state.total){
+        result = this.state.total;
+      }else {
+        if(this.state.next){
+          result = this.state.next;
+        }else {
+          result = '';
+        }
+      }
+    }*/
+    return (
+      <div className = "calculator-container">
```

Comment:
- [ optional] you can add the size of your calculator and improve on the design as well especially when a user hover over a button. great job